ui/src/contexts/Application.jsx

```diff
@@ -84,14 +97,34 @@ function watchOffers(dispatch, INSTANCE_BOARD_ID) {
         id,
         instanceHandleBoardId,
         continuingInvitation,
+        status,
       } of offers) {
         if (
           instanceHandleBoardId === INSTANCE_BOARD_ID &&
-          !watchedVaults.has(id) &&
           continuingInvitation === undefined // AdjustBalances and CloseVault offers use continuingInvitation
         ) {
-          watchedVaults.add(id);
-          watchVault(id, dispatch);
+          if (status === 'decline') {
```

Comment:
This file depends on a lot of contracts and notifiers that would make unit testing a big challenge. I would prefer not to prioritize that over fixing other issues at this point, but I'm calling this out just in case.